plugin/core/registry.py

```diff
@@ -220,3 +275,31 @@ class LspPrevDiagnosticCommand(LspTextCommand):
 
     def run(self, edit: sublime.Edit, point: Optional[int] = None) -> None:
         navigate_diagnostics(self.view, point, forward=False)
+
+
+class LspExpandTreeItemCommand(LspWindowCommand):
```

Comment:
Those commands and call-hierarchy specific but the name doesn't suggest it.
Also those have quite a bit of duplicated logic.

I'm wondering about an alternative ways to do it:
 - Hav a common base for sheet commands so that we can reuse the code to lookup the sheet at least
 - or... Have a single command for handling (call hierarchy) sheet commands that accepts an argument to decide what action within the sheet should be done.

Reply:
They are not call-hierarchy specific. I intentionally made the "TreeViewSheet" implementation as generic as possible. So that it can be reused for type-hierarchy, for example. If you want to implement the type-hierarchy request, you can probably just copy the *call_hierarchy.py* file, change a handful of lines (mainly `get_children` and `get_tree_item` methods) and it should work.

> * or... Have a single command for handling (call hierarchy) sheet commands that accepts an argument to decide what action within the sheet should be done.

Not possible/useful for the reason explained above.

Reply:
I'm talking about the `LspExpandTreeItemCommand` / `LspCollapseTreeItemCommand` commands.

Reply:
I mean you can of course combine the "expand" and "collapse" commands into a single command, and I actually wanted to do this initially, but then you need to pass another variable that you have to handle within the command.
And you can't combine it with the `LspCallHierarchyToggleCommand`, because that one is indeed call-hierarchy specific.

Reply:
Though `LspExpandTreeItemCommand` and `LspCollapseTreeItemCommand` also check for call hierarchy capability so can't be used for other things.

Reply:
> also check for call hierarchy capability

Fixed in https://github.com/sublimelsp/LSP/pull/2151/commits/ec84c88fece861dc5fa713efb37d7f226907c48e